File: csharp/HyperSwitchAI/src/ListStrategies.cs
using System;
using System.Net.Http;
using System.Text.Json;
using System.Threading.Tasks;
using DotNetEnv;

namespace HyperSwitchAI
{
    public class StrategyInfo
    {
        public string? code { get; set; }
        public string? type { get; set; }
        public string? model { get; set; }
        public string? keyId { get; set; }
        public string? provider { get; set; }
    }

    public class ListStrategiesResponse
    {
        public StrategyInfo[]? strategies { get; set; }
    }

    public class ListStrategies
    {
        private const string BASE_URL = "https://api.hyperswitchai.com";

        public static async Task RunAsync()
        {
            try
            {
                Env.Load();
                var token = await CacheToken.GetToken();

                // List all strategies
                using (var client = new HttpClient())
                {
                    client.DefaultRequestHeaders.Add("Authorization", $"Bearer {token}");
                    var response = await client.GetAsync($"{BASE_URL}/admin/strategies/list");

                    if (!response.IsSuccessStatusCode)
                    {
                        var errorText = await response.Content.ReadAsStringAsync();
                        Console.WriteLine($"\n\u001b[31mServer response: {response.StatusCode} {errorText}\u001b[0m");
                        throw new Exception($"Failed to list strategies: {response.StatusCode}");
                    }

                    var responseBody = await response.Content.ReadAsStringAsync();
                    var result = JsonSerializer.Deserialize<ListStrategiesResponse>(responseBody);

                    Console.WriteLine("\n\u001b[32m✅ Strategies listed successfully\u001b[0m");
                    Console.WriteLine(JsonSerializer.Serialize(result, new JsonSerializerOptions 
                    { 
                        WriteIndented = true 
                    }));
                }
            }
            catch (Exception ex)
            {
                Console.WriteLine($"Error: {ex.Message}");
            }
        }
    }
}

File: java/src/main/java/com/hyperswitchai/examples/ListStrategies.java
package com.hyperswitchai.examples;

import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.databind.SerializationFeature;

import java.net.URI;
import java.net.http.HttpClient;
import java.net.http.HttpRequest;
import java.net.http.HttpResponse;
import java.time.Duration;
import java.util.Map;

public class ListStrategies {
    private static final String BASE_URL = "https://api.hyperswitchai.com";
    private static final ObjectMapper mapper = new ObjectMapper()
        .enable(SerializationFeature.INDENT_OUTPUT);
    private static final HttpClient client = HttpClient.newBuilder()
        .connectTimeout(Duration.ofSeconds(10))
        .build();

    public static void main(String[] args) throws Exception {
        try {
            System.out.println("\nStarting ListStrategies...");
            
            // Get token
            String token = TokenManager.getToken();
            System.out.println("Token obtained, fetching strategies...\n");

            // Create request
            HttpRequest request = HttpRequest.newBuilder()
                .uri(URI.create(BASE_URL + "/admin/strategies/list"))
                .header("Content-Type", "application/json")
                .header("Authorization", "Bearer " + token)
                .timeout(Duration.ofSeconds(10))
                .GET()
                .build();

            // Make synchronous request
            HttpResponse<String> response = client.send(request, HttpResponse.BodyHandlers.ofString());
            
            System.out.println("Response received with status: " + response.statusCode());

            if (response.statusCode() != 200) {
                throw new RuntimeException("Failed to list strategies. Status: " + response.statusCode() + 
                    "\nResponse: " + response.body());
            }

            // Parse and display the strategies
            Map<String, Object> result = mapper.readValue(response.body(), Map.class);
            System.out.println("\n\u001B[32m✅ Current Strategies:\u001B[0m");
            System.out.println(mapper.writerWithDefaultPrettyPrinter().writeValueAsString(result));

        } catch (Exception e) {
            System.err.println("\n\u001B[31mError occurred: " + e.getMessage() + "\u001B[0m");
            throw e; // Rethrow to ensure non-zero exit code
        }
    }
} 